server/sonar-web/src/main/js/apps/about/components/AboutApp.js

```diff
@@ -0,0 +1,194 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import React from 'react';
+import keyBy from 'lodash/keyBy';
+import LoginSection from './LoginSection';
+import LoginForm from './LoginForm';
+import DropImage from './DropImage';
+import AboutProjects from './AboutProjects';
+import AboutIssues from './AboutIssues';
+import AboutScanners from './AboutScanners';
+import { translate } from '../../../helpers/l10n';
+import '../styles.css';
+import { searchProjects } from '../../../api/components';
+import { getFacet } from '../../../api/issues';
+
+const links = {
+  leak: 'http://docs.sonarqube.org/display/HOME/Fixing+the+Water+Leak',
+  qualityGates: 'http://docs.sonarqube.org/display/SONAR/Quality+Gates',
+  rules: 'http://docs.sonarqube.org/display/SONAR/Rules'
+};
+
+export default class AboutApp extends React.Component {
+  state = {
+    loading: true
+  };
+
+  componentDidMount () {
+    this.mounted = true;
+    this.loadData();
+  }
+
+  componentWillUnmount () {
+    this.mounted = false;
+  }
+
+  loadProjects () {
+    return searchProjects({ ps: 1 }).then(r => r.paging.total);
+  }
+
+  loadIssues () {
+    return getFacet({ resolved: false }, 'types').then(r => keyBy(r.facet, 'val'));
+  }
+
+  loadData () {
+    Promise.all([
+      window.sonarqube.appStarted,
+      this.loadProjects(),
+      this.loadIssues()
+    ]).then(responses => {
+      if (this.mounted) {
+        const [options, projectsCount, issueTypes] = responses;
+        this.setState({
+          projectsCount,
+          issueTypes,
+          logoUrl: options.logoUrl,
+          logoWidth: options.logoWidth,
+          loading: false
+        });
+      }
+    });
+  }
+
+  render () {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Function has a complexity of 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=javascript%3AFunctionComplexity)